tests/xtro-sharpie/EnumCheck.cs

```diff
@@ -122,25 +126,107 @@ public override void VisitEnumDecl (EnumDecl decl, VisitKind visitKind)
 			}
 
 			int managed_size = 4;
+			bool signed = true;
 			switch (GetEnumUnderlyingType (type).Name) {
 			case "Byte":
+				signed = false;
+				managed_size = 1;
+				break;
 			case "SByte":
 				managed_size = 1;
 				break;
 			case "Int16":
+				managed_size = 2;
+				break;
 			case "UInt16":
+				signed = false;
 				managed_size = 2;
 				break;
 			case "Int32":
+				break;
 			case "UInt32":
+				signed = false;
 				break;
 			case "Int64":
+				managed_size = 8;
+				break;
 			case "UInt64":
+				signed = false;
 				managed_size = 8;
 				break;
 			default:
 				throw new NotImplementedException ();
 			}
+
+			var fields = type.Fields;
+			if (signed) {
+				managed_signed_values.Clear ();
+				native_signed_values.Clear ();
+				// skip `0` since it's `__value` (special)
```

Comment:
Is the `__value` field guaranteed to always be the first field?

Reply:
It's actually `value__` and there's no order mentioning in the ECMA spec
I guess it's easier for compiler to emit the underlying size before adding members :)
It's easy to fix (along with the typo)